pkg/controller/atlas/client_test.go

```diff
@@ -17,3 +21,22 @@ func TestClientUserAgent(t *testing.T) {
 	r.NoError(err)
 	r.Contains(c.UserAgent, version.Version)
 }
+
+// RoundTrip implements http.RoundTripper registering if it got used
+type testTransport struct {
+	used bool
+}
+
+func (tt *testTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	tt.used = true
+	return nil, nil
+}
+
+func TestCustomTransport(t *testing.T) {
+	tt := &testTransport{used: false}
+
+	client, err := atlas.Client("https://cloud.mongodb.com", atlas.Connection{}, nil, httputil.CustomTransport(tt))
```

Comment:
One nit: @josvazg maybe you meant "cloud-qa"?

Reply:
right, but it does not really matter as we are not accessing anything on the network in this test, the custom transport does nothing. We could even say here `"https://invented.company.co"`
want me to change it anyway?